src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +165,999 @@ static DamageTable D_80A63CE0 = {
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_ON,
+        },
+        { TINGLE_LIMB_BALLOON, { { 2400, 0, 0 }, 50 }, 100 },
+    },
+};
 
-// static ColliderJntSphElementInit sJntSphElementsInit[1] = {
-static ColliderJntSphElementInit D_80A63D0C[1] = {
+static ColliderJntSphInit sJntSphInit = {
     {
-        { ELEMTYPE_UNK0, { 0xF7CFFFFF, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
-        { 6, { { 2400, 0, 0 }, 50 }, 100 },
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
     },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit, // sJntSphElementsInit,
+};
+
+static s16 sBuyMapOptions[6][2] = {
+    { TINGLE_MAP_CLOCK_TOWN, TINGLE_MAP_WOODFALL },   { TINGLE_MAP_WOODFALL, TINGLE_MAP_SNOWHEAD },
+    { TINGLE_MAP_SNOWHEAD, TINGLE_MAP_ROMANI_RANCH }, { TINGLE_MAP_ROMANI_RANCH, TINGLE_MAP_GREAT_BAY },
+    { TINGLE_MAP_GREAT_BAY, TINGLE_MAP_STONE_TOWER }, { TINGLE_MAP_STONE_TOWER, TINGLE_MAP_CLOCK_TOWN },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_80A63D30 = {
-    { COLTYPE_NONE, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    ARRAY_COUNT(sJntSphElementsInit), D_80A63D0C, // sJntSphElementsInit,
+static AnimationInfo sAnimationInfo[14] = {
+    /* 0 */ { &gTingleFloatIdleAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 1 */ { &gTingleFallAnim, 1.5f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 2 */ { &gTingleFallAnim, 1.5f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f },
+    /* 3 */ { &gTingleLandAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f },
+    /* 4 */ { &gTingleTalkTwistAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 5 */ { &gTingleTalkIdleAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 6 */ { &gTingleMagicSpellAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -8.0f },
+    /* 7 */ { &gTingleHappyDanceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 8 */ { &gTingleHappyDanceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -8.0f },
+    /* 9 */ { &gTingleMagicSpellAnim, 1.0f, 23.0f, 0.0f, ANIMMODE_ONCE, -8.0f },
+    /* 10 */ { &gTingleWaitAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },
+    /* 11 */ { &gTingleSpinAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+    /* 12 */ { &gTingleHideFaceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
+    /* 13 */ { &gTingleThrowConfettiAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },
 };
 
-#endif
+static Vec3f sInitBalloonScale = { 0.0f, 0.0f, 0.0f };
+
+void EnBal_Init(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
+    s32 pad;
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&gTingleFloatIdleAnim);
+    this->location = EN_BAL_GET_LOCATION(&this->picto.actor);
```

Comment:
```suggestion
    this->mapId = EN_BAL_GET_MAP_ID(&this->picto.actor);
```
Same here, the names should reflect this.

Could also change `EN_BAL_GET_MAP_ID` -> `TINGLE_GET_MAP_ID` to be more readible

Reply:
Is there a good alternate name for the field I named "location" (0x3AE)?
The problem with using "mapId" for now is that I named a different field "mapId" (0x3AC). These do appear to serve different functions - 3AE notes what "type" of Tingle (ie. where in Termina you find him) this actor instance is while 3AC notes the mapId of the map the player selected for purchase, which of course may not necessarily match Tingle's current location.
Maybe something like "mapId" and "requestedMapId" or something?

Reply:
Oh I see. Yeah cause there's the place Tingle is located, but then you have the option to either buy the map where Tingle is located or buy the map in the next location. Maybe you're right, but would like others thoughts on this as well if anyone has any ideas.

If they were to be merged into both usingi `MapId`, I don't know if it's the best names for it, but my first idea would be `baseMapId` and `boughtMapId`? Or maybe `locationMapId` and `purchasedMapId`? Just brainstorming

Reply:
`locationMapId` and `purchaseMapId` seem good to me (not too attached to dropping the `d`, but would slightly prefer just `purchase`)